### DISCUSSION TITLE
ファームウエアの読み込み失敗（RESETボタンも押せない状態）

### DISCUSSION THREAD

**Discussion by salty815** [Yowkees/keyball#201] [Q&A]
首記の通り、ファームウエアを色々いじっているうちに片一方のキーボード（左側）のキーが全く反応せず、RESETボタンを押せない状態になってしまったので助力いただきたいです。

前提
動作環境→Win11
keyball44, 左トラックボール

現象
★キーボード側差し込み口（Type-C）右側につながれた時、TRRSケーブルでつながれた左側のキー及びトラボの操作が動かない
・左側キーボードのLEDは光っていることから、TRRSケーブルの不良ではなさそう
・名称がわからないのですが、信号部（TRRSケーブル差し込み口付近）は右側キーボードのキーを押すと反応するが、左側は反応なし
・キーボード側差し込み口左側につながれた時には、どちら側のキーボードのキー及びトラボは正常に動作する

直前の操作（原因を引き起こしたであろう操作）
・QMKのビルド環境構築（QMK‐MSYS）参考URL→https://wecanpanic.hatenablog.com/entry/2023/02/05/KeyBall%E3%81%AE%E3%83%95%E3%82%A1%E3%83%BC%E3%83%A0%E3%82%A6%E3%82%A7%E3%82%A2%E3%82%92%E3%82%A2%E3%83%83%E3%83%97%E3%82%B0%E3%83%AC%E3%83%BC%E3%83%89%E3%81%99%E3%82%8B%E3%81%9E

・ビルドができているかの確認のため、以下コマンドでhexファイル作成、そのファイルをPro Micro Web Updaterでflash
make -j8 SKIP_GIT=yes keyball/keyball44:default

・RemapでKEYBOARDを読み込む際にそもそもkeyball44が認識されない
・★の現象確認。
→Pro Micro Web Updaterで正常動作保証のあるhexファイルをflashしようと考えたが、RESETボタンが左側キーボードのボタンのみで押せるマッピングとなっており、★の問題によりRESETボタンが押せない。←今ここ

何かほかに情報提供が必要であればおっしゃってください。よろしくお願いいたします。

**Comment by salty815** [comment]
![Remap](https://user-images.githubusercontent.com/59037942/226266288-19eca445-32ba-4275-888f-0e030d8f0d1a.png)

追記
keyball39でもkeyball44と同様にdefaultをビルド、flashした際にRemapで認識させたときの写真が上記です。この時はRemap側はkeyball39を読み込んでくれたのですが、写真左側のようにトラボを読み込んでない設計になっております。仮説ですがもしかするとkeyball39側でもこのような変なマッピングがされてしまっており、問題が引き起こされたのでは？と思っております。。。

**Comment by Yowkees** [comment]
リセットスイッチはTRSコネクタ近くのタクトスイッチになります。
これを正規ファームウェアの書き込み直前に2回素早く押すと、ProMicroの動作が一定時間停止して書き込めるはずです。
ご検討お願いします🙏

**Comment by salty815** [comment]
返信ありがとうございます。こちら解決いたしました。
別途ご連絡させていただきましたが、改めてこちらのスレッドで追加質問させてください。


質問
https://remap-keys.app/catalog/tAJ9Htme4oNabUkx4832/firmware
上記リンクのKeyball44_viaのDOWNLOADボタン押下時ののhexファイルと、
下記リンクのviaをビルド時に作成されるhexファイルは同一のものとしてよろしいでしょうか。

https://github.com/Yowkees/keyball/tree/main/qmk_firmware/keyboards/keyball/keyball44/keymaps


上記リンクのKeyball44_viaのSOURCE CODEボタン押下時のリンク先がkeyball44/v0.0.1-alpha.2となっており、
本家のviaファイルと違うのではないか懸念点があるため、お聞かせください。(diffを取った限り違いがあるようには見えませんが念のためお聞かせいただきたく。)

**Comment by salty815** [comment]
![Remap](https://user-images.githubusercontent.com/59037942/227759383-50e3733c-728d-47c7-a1af-2b2ffd125ef7.png)
追記
keyball44/v0.0.1-alpha.2と下記リンク先のkeyball44のマッピングでビルド、flashを行いましたがどちらでも添付ファイルのようにkeyballを読み取ってくれずです。。。（期待値は左側3つのレイヤーキー？が2つ表示される認識）
https://github.com/Yowkees/keyball/tree/main/qmk_firmware/keyboards/keyball/keyball44/keymaps
